### DISCUSSION TITLE
Getting  stream video

### DISCUSSION THREAD

**Discussion by Gerarca** [axios/axios#5882] [Q&A] [answered]
### Describe the bug

Hello!! I just tring to get video stream from camera and currently I'm getting this:

[https://dc601.4shared.com/img/2cvnV47iku/s24/1860d60e3b8/Untitled?async&rand=0.6520088914167292]([https://dc601.4shared.com/img/2cvnV47iku/s24/1860d60e3b8/Untitled?async&rand=0.6520088914167292](https://github.com/axios/axios/issues/url))

```js
//file request.js
exports.request = async ({path, method = "GET", body, token }) => {

    const httpsAgent = new https.Agent({
        rejectUnauthorized: false,
    });

    try
    {  
        const response = await axios({
            method: 'GET',
            url: path,
            responseType: 'stream',
            httpsAgent: httpsAgent,
            headers: {
                'Authorization': `Bearer ${token}`
            },
            });  

        return response;    
    }
    catch (Exception)
    {
        console.log("Exception: ", Exception);
    } 
}

//file controller.js
exports.gethttpStreaming = async (req, res) => { 
    const data = await request({
        path: urlHttpStreamming,
        method: 'StreamLive',
        body: {},
        token: req.query.token
    });

    res.set('Access-Control-Allow-Origin', '*');
    res.set('Cache-Control', 'no-cache');
    res.set('Connection', 'Keep-Alive');
    res.set('Content-Type', 'video/mp4');
    res.set('Keep-Alive', 'timeout=5');
    res.set('Transfer-Encoding', 'chunked'); 
    data.data.pipe(res);
}
```

As you can see I'm setting the header, in such a way that it will receive the streaming video, what can I do? 

### Axios Version

1.2.2

### Browser

chrome

### Browser Version

109.0.5414.120

### Node.js Version

18.12.1

### OS

windows 11

### Additional context/Screenshots

```bash
https://dc601.4shared.com/img/2cvnV47iku/s24/1860d60e3b8/Untitled?async&rand=0.6520088914167292
```

**Comment by jasonsaayman** [comment, marked as answer]
`responseType: 'stream'` is Node-only and your Axios side looks mostly correct: `data.data` is the readable stream, so piping it to `res` is the right idea.

A few things to fix/check:
```js
const upstream = await request({ path: urlHttpStreamming, token: req.query.token });

res.status(upstream.status);

for (const [key, value] of Object.entries(upstream.headers)) {
  if (value != null) {
    res.setHeader(key, value);
  }
}

upstream.data.on('error', (err) => {
  res.destroy(err);
});

upstream.data.pipe(res);
```

Avoid manually setting Transfer-Encoding, Connection, and Keep-Alive; Node will manage those. Also make sure the upstream stream is actually MP4. Many camera streams are MJPEG, HLS, RTSP, or another format, and setting Content-Type: video/mp4 will not convert the stream.

One security note: rejectUnauthorized: false disables TLS certificate validation and should not be used in production.
